src/v/cluster/partition.h

```diff
@@ -345,8 +345,16 @@ class partition : public ss::enable_lw_shared_from_this<partition> {
 
     bool may_read_from_cloud() const;
 
-    ss::future<std::optional<storage::timequery_result>>
-      local_timequery(storage::timequery_config);
+    using fallback_mode = ss::bool_class<struct fallback_mode_tag>;
```

Comment:
Is it really necessary to use bool_class here? It's not part of the public interface.